src/countdown/Countdown.tsx

```diff
@@ -0,0 +1,66 @@
+import "./_countdown.scss";
+
+import React, {useLayoutEffect} from "react";
+import classNames from "classnames";
+
+import useCountdownTimer from "../core/utils/hooks/useCountdownTimer";
+import List from "../list/List";
+import ListItem from "../list/item/ListItem";
+import {generateCountdownItems} from "../core/utils/time/timeUtils";
+
+export interface CountdownProps {
+  testid: string;
+  startDate: Date;
+  countDownIntervalInSeconds?: number;
+  alwaysShowSeconds?: boolean;
+  onEnd?: () => void;
+  customClassName?: string;
+}
+
+export interface CountdownItem {
+  id: "days" | "hours" | "minutes" | "seconds";
+  count: string | number;
+}
+
+function Countdown({
+  testid,
+  startDate,
+  countDownIntervalInSeconds = 1,
+  alwaysShowSeconds = false,
+  onEnd,
+  customClassName
+}: CountdownProps) {
+  const countdownData = useCountdownTimer(startDate, countDownIntervalInSeconds);
+  let items = generateCountdownItems(countdownData, alwaysShowSeconds);
+
+  useLayoutEffect(() => {
+    if (countdownData.delta <= 0 && onEnd) {
+      onEnd();
+    }
+  }, [countdownData.delta, onEnd]);
```

Comment:
We can't assume that the users of this component will remember to use `useCallback` when defining their `onEnd` handler. Let's make sure we save the callback and then use it within the effect

```
const savedOnEndCallback = useRef<CountdownProps["onEnd"]>();

useLayoutEffect(() => {
   savedOnEndCallback.current = onEnd;
});
```

Then use `savedOnEndCallback.current` within the layout effect that checks `countdownData.delta`